src/Foreign/Index.purs

```diff
@@ -17,37 +17,36 @@ import Prelude
 
 import Control.Monad.Except.Trans (ExceptT)
 
-import Foreign (Foreign, F, ForeignError(..), typeOf, isUndefined, isNull, fail)
+import Foreign (Foreign, FT, ForeignError(..), typeOf, isUndefined, isNull, fail)
 import Data.Function.Uncurried (Fn2, runFn2, Fn4, runFn4)
-import Data.Identity (Identity)
 import Data.List.NonEmpty (NonEmptyList)
 
 -- | This type class identifies types that act like _property indices_.
 -- |
 -- | The canonical instances are for `String`s and `Int`s.
-class Index i where
-  index :: Foreign -> i -> F Foreign
+class Index i m | i -> m where
+  index :: Foreign -> i -> FT m Foreign
```

Comment:
Instead of adding the `m` to the class here, perhaps we can just change the type of `index` / `ix`:

``` purescript
  index :: forall m. Monad m => Foreign -> i -> FT m Foreign
```
?

Reply:
I’m not sure this is possible because of the instance of `Indexable` for `ExceptT (NonEmptyList ForeignError) m Foreign`. Removing the functional dependency and quantifying `m` in `index` yields the following error:

```
[1/1 TypesDoNotUnify] src/Foreign/Index.purs:84:12

  84    ix f i = flip index i =<< f
                 ^^^^^^^^^^^^^^^^^^
  
  Could not match type
  
    m1
  
  with type
  
    m00
  
  while trying to match type ExceptT (NonEmptyList ForeignError) m1
    with type ExceptT (NonEmptyList ForeignError) m00
  while checking that expression (bindFlipped ((flip index) i)) f
    has type ExceptT (NonEmptyList ForeignError) m00 Foreign
  in value declaration indexableExceptT
  
  where m1 is a rigid type variable
          bound at (line 0, column 0 - line 0, column 0)
        m00 is a rigid type variable
          bound at (line 0, column 0 - line 0, column 0)

Reply:
Ah yeah, I see now! Withdrawing this then.